GSK-342/task Added the case where prediction is constant after perturbation

rabah-khalek opened the pull request:
`stats.wilcoxon(population_1, population_2, alternative=alternative)` was failing when `population_1==population_2`. I added an if statement to avoid this issue.

andreybavt commented on giskard/ml_worker/testing/stat_utils.py:
```diff
@@ -14,7 +15,10 @@ def paired_t_test(population_1, population_2, alternative, critical_quantile):
+    if np.array_equal(population_1, population_2):
+        p_value = 1.
+    else:
+        stat, p_value = stats.ttest_rel(population_1, population_2, alternative=alternative)
```
```suggestion
        _, p_value = stats.ttest_rel(population_1, population_2, alternative=alternative)
```

it's common to replace unused variables with `_`

andreybavt commented on giskard/ml_worker/testing/stat_utils.py:
```diff
@@ -14,7 +15,10 @@ def paired_t_test(population_1, population_2, alternative, critical_quantile):
+    else:
+        stat, p_value = stats.ttest_rel(population_1, population_2, alternative=alternative)
 
     if p_value > critical_quantile:
```
this whole if-else block can be just 
```
return p_value <= critical_quantile, p_value
```

andreybavt reviewed (changes requested):
@rabah-khalek could you implement unit tests for the new code?

rabah-khalek commented:
@andreybavt, I implemented all your suggestion and I:
- re-debugged all the stat tests after the modifications, all cases are working as expected.
- I splitted the unit tests into direction and invariance based on their case (they were all in invariance before).
- removed some overlooked `return` in unit tests
- improvied the coverage of giskard/ml_worker/testing/stat_utils.py up to 95%.
